src/proteus/dswx_hls.py

```diff
@@ -2933,9 +3057,10 @@ def _create_ocean_mask(shape_file, margin_km, scratch_dir,
             # Set up the shapefile driver 
             shapefile_driver = ogr.GetDriverByName("ESRI Shapefile")
 
-            # TODO fix filename:
             temp_shapefile_filename = tempfile.NamedTemporaryFile(
                 dir=scratch_dir, suffix='.shp').name
+            if temp_files_list is not None:
+                temp_files_list.append(temp_shapefile_filename)
```

Comment:
For some reason, there are still three additional temp files created in addition to the .shp temp file. I don't know where/why/how these additional temp files are created, but their names appear to be consistent with the .shp file's name. This is a hack that seems to fix the problem:

```suggestion
                temp_files_list.append(temp_shapefile_filename)
                temp_files_list.append(temp_shapefile_filename.replace('.shp','.prj'))
                temp_files_list.append(temp_shapefile_filename.replace('.shp','.dbf'))
                temp_files_list.append(temp_shapefile_filename.replace('.shp','.shx'))
```

Reply:
Thank you for catching that, @nemo794